orchestrator/lib/coolfhir/error.go

```diff
@@ -97,15 +97,15 @@ func WriteOperationOutcomeFromError(err error, desc string, httpResponse http.Re
 	var operationOutcome fhir.OperationOutcome
 
 	// Error type: fhirclient.OperationOutcomeError
-	var operationOutcomeErr *fhirclient.OperationOutcomeError
-	if errors.As(err, &operationOutcomeErr) {
+	var operationOutcomeErr = new(fhirclient.OperationOutcomeError)
+	if errors.As(err, operationOutcomeErr) || errors.As(err, &operationOutcomeErr) {
 		if operationOutcomeErr.HttpStatusCode > 0 {
 			statusCode = operationOutcomeErr.HttpStatusCode
 		}
 		operationOutcome = operationOutcomeErr.OperationOutcome
 	} else {
 		// Error type: ErrorWithCode
-		var errorWithCode *ErrorWithCode
+		var errorWithCode = new(ErrorWithCode)
 		if errors.As(err, &errorWithCode) {
```

Comment:
```suggestion
		if errors.As(err, errorWithCode) || errors.As(err, &errorWithCode) {
```
For parity with the above check